pkg/controller/util/node/controller_utils.go

```diff
@@ -173,18 +173,27 @@ func MarkAllPodsNotReady(kubeClient clientset.Interface, node *v1.Node) error {
 
 // ExistsInCloudProvider returns true if the node exists in the
 // cloud provider.
-func ExistsInCloudProvider(cloud cloudprovider.Interface, nodeName types.NodeName) (bool, error) {
+func ExistsInCloudProvider(cloud cloudprovider.Interface, providerID string, nodeName types.NodeName) (bool, error) {
 	instances, ok := cloud.Instances()
 	if !ok {
 		return false, fmt.Errorf("%v", ErrCloudInstance)
 	}
-	if _, err := instances.ExternalID(context.TODO(), nodeName); err != nil {
+
+	exists, err := instances.InstanceExistsByProviderID(context.TODO(), providerID)
```

Comment:
</picky> Why not just use providerIDErr here and skip the error shuffle later?

Reply:
This is a copy/pasta of the cloud node controller. I plan to move the cloud node controller to use this method as well in a separate PR so we avoid this in the future hopefully.